### PR TITLE
add (auto)chunk and its first module llama_index_chunk

### PR DESCRIPTION
**Pull request by bwook00**
Plan to write docs after code review ^^

### CODE REVIEW

File: tests/autorag/data/chunk/test_chunk_base.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+root_dir = pathlib.PurePath(
+	os.path.dirname(os.path.realpath(__file__))
+).parent.parent.parent
+resource_dir = os.path.join(root_dir, "resources")
+data_dir = os.path.join(resource_dir, "chunk_data")
+
+base_texts = [
+	"The Kia Tigers lost the Korean Series this year and failed to win the championship. jeffrey went to "
+	"gwangju to the Korean Series, but they lost there. I love this story.",
```
Comment (vkehfdl1) [inline tests/autorag/data/chunk/test_chunk_base.py:16]: I hate this story

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]
으악

File: autorag/data/chunk/base.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			kwargs.update({"sentence_splitter": sentence_splitter})
+
+	# Add embed_model to kwargs
+	embedding_available_methods = ["semantic_llama_index"]
+	if chunk_method in embedding_available_methods:
+		# there is no default embed_model, so we have to get it parameter and add it.
+		if "embed_model" not in kwargs.keys():
+			raise ValueError(f"embed_model is required for {chunk_method} method.")
+		embed_model_str = kwargs.pop("embed_model")
+		embed_model = embedding_models[embed_model_str]()
```
[the remark this answers is not in the document]
Comment (bwook00) [reply autorag/data/chunk/base.py:83]: If i delete `()`, `type(embed_model)` is `autorag.LazyInit`
but, if i have `()` embed_model's type is `llama_index.embeddings.openai.base.OpenAIEmbedding`

For some reason, LazyInit's `def __call__(self):` requires a () to be used.

File: autorag/data/chunk/llama_index_chunk.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	# make metadata
+	metadata = list(
+		map(
+			lambda node: add_essential_metadata_llama_text_node(
+				node.metadata, node.relationships
+			),
+			chunk_results,
+		)
+	)
+	return doc_id, contents, metadata
```
[the remark this answers is not in the document]
Comment (bwook00) [reply autorag/data/chunk/llama_index_chunk.py:48]: good idea. I'll do it issue #684

File: autorag/data/chunk/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vkehfdl1) [reply autorag/data/chunk/base.py:83]: Ah okay I got it

**Review by vkehfdl1 — APPROVED** [review]
LGTM

**Pull request merged**